### PR TITLE
feat: ssl for postgres destinations

### PR DESCRIPTION
**Pull request by lucasmarshall**
This needs some testing

### CODE REVIEW

File: apps/mgmt-ui/src/components/configuration/destination/PostgresDestinationDetailsPane.tsx [hunk 1]
```diff
@@ -198,6 +205,18 @@ export default function PostgresDestinationDetailsPanel({ isLoading }: PostgresD
           />
         </Stack>
 
+        <Stack className="gap-2">
+          <Typography variant="subtitle1">SSL</Typography>
+          <Select
```
Comment (asdfryan) [inline apps/mgmt-ui/src/components/configuration/destination/PostgresDestinationDetailsPane.tsx:210]: Should this be a checkbox instead?

Comment (lucasmarshall) [reply apps/mgmt-ui/src/components/configuration/destination/PostgresDestinationDetailsPane.tsx:210]: There probably should be other options down the line.

**Review by asdfryan — COMMENTED** [review]
LGTM modulo build failures

File: apps/mgmt-ui/src/components/configuration/destination/PostgresDestinationDetailsPane.tsx [hunk 2]
```diff
@@ -58,6 +65,23 @@ export default function PostgresDestinationDetailsPanel({ isLoading }: PostgresD
...
+    if (!ca) {
+      setCa(destination.config.ca);
+    }
+    if (!cert) {
+      setCert(destination.config.cert);
+    }
+    if (!key) {
+      setKey(destination.config.key);
+    }
+    if (!passphrase) {
```
Comment (tomkit) [inline apps/mgmt-ui/src/components/configuration/destination/PostgresDestinationDetailsPane.tsx:80]: In another PR I removed these `if()` guards now that the 2nd argument (`dependencies`) for it is working. Having the `if()` guards would mean some values are set, but not others based on the value being truthy/falsy

File: openapi/mgmt/components/schemas/objects/postgres_config.yaml [hunk 3]
```diff
@@ -18,10 +18,41 @@ properties:
...
+    example: password
+
 required:
   - host
   - port
   - database
   - schema
   - user
   - password
+  - sslmode
```
Comment (tomkit) [inline openapi/mgmt/components/schemas/objects/postgres_config.yaml:53]: We might want to add a lightweight of differentiating between different types of connection configurations, e.g. not all will require certificates

**Review by asdfryan — APPROVED** [review]

**Pull request closed without merging**